// @webdatarocks/ng-webdatarocks/__ivy_ngcc__/fesm2015/webdatarocks-ng-webdatarocks.js
import { Component, ElementRef, Input, Output, EventEmitter, NgModule } from '@angular/core';
import * as WebDataRocks from '@webdatarocks/webdatarocks';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';
class WebdatarocksComponent {
    /**
     * @param {?} el
     */
    constructor(el) {
        this.el = el;
        // events
        this.cellclick = new EventEmitter();
        this.celldoubleclick = new EventEmitter();
        this.dataerror = new EventEmitter();
        this.datafilecancelled = new EventEmitter();
        this.dataloaded = new EventEmitter();
        this.datachanged = new EventEmitter();
        this.fieldslistclose = new EventEmitter();
        this.fieldslistopen = new EventEmitter();
        this.filteropen = new EventEmitter();
        this.fullscreen = new EventEmitter();
        this.loadingdata = new EventEmitter();
        this.loadinglocalization = new EventEmitter();
        this.loadingreportfile = new EventEmitter();
        this.localizationerror = new EventEmitter();
        this.localizationloaded = new EventEmitter();
        this.openingreportfile = new EventEmitter();
        this.querycomplete = new EventEmitter();
        this.queryerror = new EventEmitter();
        this.ready = new EventEmitter();
        this.reportchange = new EventEmitter();
        this.reportcomplete = new EventEmitter();
        this.reportfilecancelled = new EventEmitter();
        this.reportfileerror = new EventEmitter();
        this.reportfileloaded = new EventEmitter();
        this.runningquery = new EventEmitter();
        this.update = new EventEmitter();
        this.beforetoolbarcreated = new EventEmitter();
        this.aftergriddraw = new EventEmitter();
        this.beforegriddraw = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.root = (/** @type {?} */ (this.el.nativeElement));
        this.webDataRocks = new WebDataRocks({
            container: this.root.getElementsByClassName('wbr-ng-wrapper')[0],
            width: this.width,
            height: this.height,
            toolbar: this.toolbar,
            report: this.report,
            global: this.global,
            customizeCell: this.customizeCell,
            cellclick: (cell) => this.cellclick.next(cell),
            celldoubleclick: (cell) => this.celldoubleclick.next(cell),
            dataerror: (event) => this.dataerror.next(event),
            datafilecancelled: () => this.datafilecancelled.next(),
            dataloaded: () => this.dataloaded.next(),
            datachanged: (event) => this.datachanged.next(event),
            fieldslistclose: () => this.fieldslistclose.next(),
            fieldslistopen: () => this.fieldslistopen.next(),
            filteropen: () => this.filteropen.next(),
            loadingdata: () => this.loadingdata.next(),
            loadinglocalization: () => this.loadinglocalization.next(),
            loadingreportfile: () => this.loadingreportfile.next(),
            localizationerror: () => this.localizationerror.next(),
            localizationloaded: () => this.localizationloaded.next(),
            openingreportfile: () => this.openingreportfile.next(),
            querycomplete: () => this.querycomplete.next(),
            queryerror: () => this.queryerror.next(),
            ready: () => this.ready.next(this.webDataRocks),
            reportchange: () => this.reportchange.next(),
            reportcomplete: () => this.reportcomplete.next(),
            reportfilecancelled: () => this.reportfilecancelled.next(),
            reportfileerror: () => this.reportfileerror.next(),
            reportfileloaded: () => this.reportfileloaded.next(),
            runningquery: () => this.runningquery.next(),
            update: () => this.update.next(),
            beforetoolbarcreated: (toolbar) => this.beforetoolbarcreated.next(toolbar),
            aftergriddraw: (event) => this.aftergriddraw.next(event),
            beforegriddraw: (event) => this.beforegriddraw.next(event)
        });
    }
}
WebdatarocksComponent.ɵfac = function WebdatarocksComponent_Factory(t) { return new (t || WebdatarocksComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
WebdatarocksComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: WebdatarocksComponent, selectors: [["app-wbr-pivot"]], inputs: { toolbar: "toolbar", width: "width", height: "height", report: "report", global: "global", customizeCell: "customizeCell" }, outputs: { cellclick: "cellclick", celldoubleclick: "celldoubleclick", dataerror: "dataerror", datafilecancelled: "datafilecancelled", dataloaded: "dataloaded", datachanged: "datachanged", fieldslistclose: "fieldslistclose", fieldslistopen: "fieldslistopen", filteropen: "filteropen", fullscreen: "fullscreen", loadingdata: "loadingdata", loadinglocalization: "loadinglocalization", loadingreportfile: "loadingreportfile", localizationerror: "localizationerror", localizationloaded: "localizationloaded", openingreportfile: "openingreportfile", querycomplete: "querycomplete", queryerror: "queryerror", ready: "ready", reportchange: "reportchange", reportcomplete: "reportcomplete", reportfilecancelled: "reportfilecancelled", reportfileerror: "reportfileerror", reportfileloaded: "reportfileloaded", runningquery: "runningquery", update: "update", beforetoolbarcreated: "beforetoolbarcreated", aftergriddraw: "aftergriddraw", beforegriddraw: "beforegriddraw" }, decls: 2, vars: 0, consts: [[1, "wbr-ng-wrapper"]], template: function WebdatarocksComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelement(1, "div", 0);
        ɵngcc0.ɵɵelementEnd();
    } }, encapsulation: 2 });
WebdatarocksComponent.ctorParameters = () => [
    { type: ElementRef }
];
WebdatarocksComponent.propDecorators = {
    toolbar: [{ type: Input }],
    width: [{ type: Input }],
    height: [{ type: Input }],
    report: [{ type: Input }],
    global: [{ type: Input }],
    customizeCell: [{ type: Input }],
    cellclick: [{ type: Output }],
    celldoubleclick: [{ type: Output }],
    dataerror: [{ type: Output }],
    datafilecancelled: [{ type: Output }],
    dataloaded: [{ type: Output }],
    datachanged: [{ type: Output }],
    fieldslistclose: [{ type: Output }],
    fieldslistopen: [{ type: Output }],
    filteropen: [{ type: Output }],
    fullscreen: [{ type: Output }],
    loadingdata: [{ type: Output }],
    loadinglocalization: [{ type: Output }],
    loadingreportfile: [{ type: Output }],
    localizationerror: [{ type: Output }],
    localizationloaded: [{ type: Output }],
    openingreportfile: [{ type: Output }],
    querycomplete: [{ type: Output }],
    queryerror: [{ type: Output }],
    ready: [{ type: Output }],
    reportchange: [{ type: Output }],
    reportcomplete: [{ type: Output }],
    reportfilecancelled: [{ type: Output }],
    reportfileerror: [{ type: Output }],
    reportfileloaded: [{ type: Output }],
    runningquery: [{ type: Output }],
    update: [{ type: Output }],
    beforetoolbarcreated: [{ type: Output }],
    aftergriddraw: [{ type: Output }],
    beforegriddraw: [{ type: Output }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(WebdatarocksComponent, [{
        type: Component,
        args: [{
                selector: 'app-wbr-pivot',
                template: `<div><div class='wbr-ng-wrapper'></div></div>`
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { cellclick: [{
            type: Output
        }], celldoubleclick: [{
            type: Output
        }], dataerror: [{
            type: Output
        }], datafilecancelled: [{
            type: Output
        }], dataloaded: [{
            type: Output
        }], datachanged: [{
            type: Output
        }], fieldslistclose: [{
            type: Output
        }], fieldslistopen: [{
            type: Output
        }], filteropen: [{
            type: Output
        }], fullscreen: [{
            type: Output
        }], loadingdata: [{
            type: Output
        }], loadinglocalization: [{
            type: Output
        }], loadingreportfile: [{
            type: Output
        }], localizationerror: [{
            type: Output
        }], localizationloaded: [{
            type: Output
        }], openingreportfile: [{
            type: Output
        }], querycomplete: [{
            type: Output
        }], queryerror: [{
            type: Output
        }], ready: [{
            type: Output
        }], reportchange: [{
            type: Output
        }], reportcomplete: [{
            type: Output
        }], reportfilecancelled: [{
            type: Output
        }], reportfileerror: [{
            type: Output
        }], reportfileloaded: [{
            type: Output
        }], runningquery: [{
            type: Output
        }], update: [{
            type: Output
        }], beforetoolbarcreated: [{
            type: Output
        }], aftergriddraw: [{
            type: Output
        }], beforegriddraw: [{
            type: Output
        }], toolbar: [{
            type: Input
        }], width: [{
            type: Input
        }], height: [{
            type: Input
        }], report: [{
            type: Input
        }], global: [{
            type: Input
        }], customizeCell: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
class WebdatarocksPivotModule {
}
WebdatarocksPivotModule.ɵfac = function WebdatarocksPivotModule_Factory(t) { return new (t || WebdatarocksPivotModule)(); };
WebdatarocksPivotModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: WebdatarocksPivotModule });
WebdatarocksPivotModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(WebdatarocksPivotModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [WebdatarocksComponent],
                exports: [WebdatarocksComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WebdatarocksPivotModule, { declarations: function () { return [WebdatarocksComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [WebdatarocksComponent]; } }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */

export { WebdatarocksPivotModule, WebdatarocksComponent };
