// @webdatarocks/ng-webdatarocks/bundles/webdatarocks-ng-webdatarocks.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@webdatarocks/webdatarocks'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('@webdatarocks/ng-webdatarocks', ['exports', '@angular/core', '@webdatarocks/webdatarocks', '@angular/common'], factory) :
    (factory((global.webdatarocks = global.webdatarocks || {}, global.webdatarocks['ng-webdatarocks'] = {}),global.ng.core,null,global.ng.common));
}(this, (function (exports,core,WebDataRocks,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var WebdatarocksComponent = /** @class */ (function () {
        function WebdatarocksComponent(el) {
            this.el = el;
            // events
            this.cellclick = new core.EventEmitter();
            this.celldoubleclick = new core.EventEmitter();
            this.dataerror = new core.EventEmitter();
            this.datafilecancelled = new core.EventEmitter();
            this.dataloaded = new core.EventEmitter();
            this.datachanged = new core.EventEmitter();
            this.fieldslistclose = new core.EventEmitter();
            this.fieldslistopen = new core.EventEmitter();
            this.filteropen = new core.EventEmitter();
            this.fullscreen = new core.EventEmitter();
            this.loadingdata = new core.EventEmitter();
            this.loadinglocalization = new core.EventEmitter();
            this.loadingreportfile = new core.EventEmitter();
            this.localizationerror = new core.EventEmitter();
            this.localizationloaded = new core.EventEmitter();
            this.openingreportfile = new core.EventEmitter();
            this.querycomplete = new core.EventEmitter();
            this.queryerror = new core.EventEmitter();
            this.ready = new core.EventEmitter();
            this.reportchange = new core.EventEmitter();
            this.reportcomplete = new core.EventEmitter();
            this.reportfilecancelled = new core.EventEmitter();
            this.reportfileerror = new core.EventEmitter();
            this.reportfileloaded = new core.EventEmitter();
            this.runningquery = new core.EventEmitter();
            this.update = new core.EventEmitter();
            this.beforetoolbarcreated = new core.EventEmitter();
            this.aftergriddraw = new core.EventEmitter();
            this.beforegriddraw = new core.EventEmitter();
        }
        /**
         * @return {?}
         */
        WebdatarocksComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                var _this = this;
                this.root = ( /** @type {?} */(this.el.nativeElement));
                this.webDataRocks = new WebDataRocks({
                    container: this.root.getElementsByClassName('wbr-ng-wrapper')[0],
                    width: this.width,
                    height: this.height,
                    toolbar: this.toolbar,
                    report: this.report,
                    global: this.global,
                    customizeCell: this.customizeCell,
                    cellclick: function (cell) { return _this.cellclick.next(cell); },
                    celldoubleclick: function (cell) { return _this.celldoubleclick.next(cell); },
                    dataerror: function (event) { return _this.dataerror.next(event); },
                    datafilecancelled: function () { return _this.datafilecancelled.next(); },
                    dataloaded: function () { return _this.dataloaded.next(); },
                    datachanged: function (event) { return _this.datachanged.next(event); },
                    fieldslistclose: function () { return _this.fieldslistclose.next(); },
                    fieldslistopen: function () { return _this.fieldslistopen.next(); },
                    filteropen: function () { return _this.filteropen.next(); },
                    loadingdata: function () { return _this.loadingdata.next(); },
                    loadinglocalization: function () { return _this.loadinglocalization.next(); },
                    loadingreportfile: function () { return _this.loadingreportfile.next(); },
                    localizationerror: function () { return _this.localizationerror.next(); },
                    localizationloaded: function () { return _this.localizationloaded.next(); },
                    openingreportfile: function () { return _this.openingreportfile.next(); },
                    querycomplete: function () { return _this.querycomplete.next(); },
                    queryerror: function () { return _this.queryerror.next(); },
                    ready: function () { return _this.ready.next(_this.webDataRocks); },
                    reportchange: function () { return _this.reportchange.next(); },
                    reportcomplete: function () { return _this.reportcomplete.next(); },
                    reportfilecancelled: function () { return _this.reportfilecancelled.next(); },
                    reportfileerror: function () { return _this.reportfileerror.next(); },
                    reportfileloaded: function () { return _this.reportfileloaded.next(); },
                    runningquery: function () { return _this.runningquery.next(); },
                    update: function () { return _this.update.next(); },
                    beforetoolbarcreated: function (toolbar) { return _this.beforetoolbarcreated.next(toolbar); },
                    aftergriddraw: function (event) { return _this.aftergriddraw.next(event); },
                    beforegriddraw: function (event) { return _this.beforegriddraw.next(event); }
                });
            };
        WebdatarocksComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'app-wbr-pivot',
                        template: "<div><div class='wbr-ng-wrapper'></div></div>"
                    },] },
        ];
        WebdatarocksComponent.ctorParameters = function () {
            return [
                { type: core.ElementRef }
            ];
        };
        WebdatarocksComponent.propDecorators = {
            toolbar: [{ type: core.Input }],
            width: [{ type: core.Input }],
            height: [{ type: core.Input }],
            report: [{ type: core.Input }],
            global: [{ type: core.Input }],
            customizeCell: [{ type: core.Input }],
            cellclick: [{ type: core.Output }],
            celldoubleclick: [{ type: core.Output }],
            dataerror: [{ type: core.Output }],
            datafilecancelled: [{ type: core.Output }],
            dataloaded: [{ type: core.Output }],
            datachanged: [{ type: core.Output }],
            fieldslistclose: [{ type: core.Output }],
            fieldslistopen: [{ type: core.Output }],
            filteropen: [{ type: core.Output }],
            fullscreen: [{ type: core.Output }],
            loadingdata: [{ type: core.Output }],
            loadinglocalization: [{ type: core.Output }],
            loadingreportfile: [{ type: core.Output }],
            localizationerror: [{ type: core.Output }],
            localizationloaded: [{ type: core.Output }],
            openingreportfile: [{ type: core.Output }],
            querycomplete: [{ type: core.Output }],
            queryerror: [{ type: core.Output }],
            ready: [{ type: core.Output }],
            reportchange: [{ type: core.Output }],
            reportcomplete: [{ type: core.Output }],
            reportfilecancelled: [{ type: core.Output }],
            reportfileerror: [{ type: core.Output }],
            reportfileloaded: [{ type: core.Output }],
            runningquery: [{ type: core.Output }],
            update: [{ type: core.Output }],
            beforetoolbarcreated: [{ type: core.Output }],
            aftergriddraw: [{ type: core.Output }],
            beforegriddraw: [{ type: core.Output }]
        };
        return WebdatarocksComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */
    var WebdatarocksPivotModule = /** @class */ (function () {
        function WebdatarocksPivotModule() {
        }
        WebdatarocksPivotModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        declarations: [WebdatarocksComponent],
                        exports: [WebdatarocksComponent]
                    },] },
        ];
        return WebdatarocksPivotModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
     */

    exports.WebdatarocksPivotModule = WebdatarocksPivotModule;
    exports.WebdatarocksComponent = WebdatarocksComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
